File: gui/src/main/java/org/correomqtt/GuiCore.java
package org.correomqtt;

import javafx.application.HostServices;
import lombok.Getter;
import org.correomqtt.core.CoreManager;
import org.correomqtt.di.SoyEvents;
import org.correomqtt.core.fileprovider.HistoryManager;
import org.correomqtt.core.plugin.PluginManager;
import org.correomqtt.core.settings.SettingsManager;
import org.correomqtt.core.utils.ConnectionManager;
import org.correomqtt.di.Inject;
import org.correomqtt.di.SingletonBean;
import org.correomqtt.gui.keyring.KeyringManager;
import org.correomqtt.gui.theme.ThemeManager;

@Getter
@SingletonBean
public class GuiCore {

    private final ConnectionManager connectionManager;
    private final SettingsManager settingsManager;
    private final HistoryManager historyManager;
    private final PluginManager pluginManager;
    private final KeyringManager keyringManager;
    private final SoyEvents soyEvents;
    private final ThemeManager themeManager;
    private final HostServices hostServices;

    @Inject
    public GuiCore(CoreManager coreManager,
                   KeyringManager keyringManager,
                   SoyEvents soyEvents,
                   ThemeManager themeManager,
                   HostServicesWrapper hostServicesWrapper) {
        this.connectionManager = coreManager.getConnectionManager();
        this.settingsManager = coreManager.getSettingsManager();
        this.historyManager = coreManager.getHistoryManager();
        this.pluginManager = coreManager.getPluginManager();
        this.keyringManager = keyringManager;
        this.soyEvents = soyEvents;
        this.themeManager = themeManager;
        this.hostServices = hostServicesWrapper.getHostServices();
    }
}

File: intellij-plugin/src/main/kotlin/com/correomqtt/plugin/GuiCore.kt
package com.correomqtt.plugin

import org.correomqtt.core.CoreManager
import org.correomqtt.core.connection.ConnectionLifecycleTaskFactories
import org.correomqtt.core.fileprovider.HistoryManager
import org.correomqtt.core.pubsub.PublishTaskFactory
import org.correomqtt.core.pubsub.SubscribeTaskFactory
import org.correomqtt.core.settings.SettingsManager
import org.correomqtt.core.utils.ConnectionManager
import org.correomqtt.di.Inject
import org.correomqtt.di.SingletonBean

/**
 * GuiCore collects all relevant Correo backend services
 * and exposes them to the plugin components / services.
 */
@SingletonBean
class GuiCore @Inject constructor(
    private var coreManager: CoreManager,
    private var connectionLifecycleTaskFactories: ConnectionLifecycleTaskFactories,
    private var publishTaskFactory: PublishTaskFactory,
    private var subscribeTaskFactory: SubscribeTaskFactory,
) {

    private var connectionManager: ConnectionManager = coreManager.connectionManager
    private var settingsManager: SettingsManager = coreManager.settingsManager
    private var historyManager: HistoryManager = coreManager.historyManager

    fun getConnectionManager(): ConnectionManager {
        return this.connectionManager;
    }

    fun getSettingsManager(): SettingsManager {
        return this.settingsManager;
    }

    fun getHistoryManager(): HistoryManager {
        return this.historyManager;
    }

    fun getSubscriptionManager(): SubscribeTaskFactory {
        return this.subscribeTaskFactory;
    }

    fun getPublishTaskManager(): PublishTaskFactory {
        return this.publishTaskFactory;
    }

    fun getConnectionLifecycleTaskFactory(): ConnectionLifecycleTaskFactories {
        return this.connectionLifecycleTaskFactories;
    }
}